### PR TITLE
Deprecate InlineMessage and old Notification components

### PR DESCRIPTION
**Pull request by amelako**
## Purpose

The brand new Notification components were released. The remaining work was to deprecated the old Notification components.

## Approach and changes

- Update the migration guide with the changes to the Notification components.
- Deprecate InlineMessage, Notification, NotificationList and NotificationCard components.

## Definition of done

* [ ] Development completed
* [ ] Reviewers assigned
* [ ] Unit and integration tests
* [ ] Meets minimum browser support
* [ ] Meets accessibility requirements

### CODE REVIEW

File: MIGRATION.md [hunk 1]
```diff
@@ -70,6 +71,38 @@ Tip: Provide the `--transform`/`-t` argument at the end of the command, so that
 > ./node_modules/.bin/circuit-ui migrate -l JavaScript -l TypeScript -t codemod-name
 > ```
 
+## From v4 to v5
+
```
Comment (robinmetral) [inline MIGRATION.md:75]: Can we add a note here about the v5 not being out yet, but adding that the draft migration guide is a good way to handle deprecations early?

**Review by robinmetral — COMMENTED** [review]
Nice! Left some thoughts and some nitpicks (sorry 🤓)

File: .changeset/honest-taxis-wave.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@sumup/circuit-ui': minor
+---
+
+Deprecated `InlineMessage`, `Notification`, `NotificationList` and `NotificationCard` components.
```
Comment (connor-baer) [inline .changeset/honest-taxis-wave.md:5]:
```suggestion
Deprecated the `InlineMessage`, `Notification`, `NotificationList`, and `NotificationCard` components. Use the new `NotificationBanner`, `NotificationInline`, and `NotificationToast` components instead.
```

File: packages/circuit-ui/components/NotificationCard/NotificationCard.tsx [hunk 3]
```diff
@@ -41,13 +42,27 @@ const innerStyles = ({ theme }: StyleProps) => css`
...
-);
+}: NotificationCardProps): JSX.Element => {
+  if (
+    process.env.NODE_ENV !== 'production' &&
+    process.env.NODE_ENV !== 'test'
+  ) {
+    deprecate(
+      'NotificationCard',
+      'The NotificationCard component is deprecated.',
+      'Use the new Notification components instead',
```
Comment (connor-baer) [inline packages/circuit-ui/components/NotificationCard/NotificationCard.tsx:59]:
```suggestion
      'Use the new notification components instead.',
```

File: packages/circuit-ui/components/NotificationList/NotificationList.tsx [hunk 4]
```diff
@@ -65,15 +66,29 @@ const cardStyles = ({ theme }: StyleProps) => css`
...
-);
+}: NotificationListProps): JSX.Element => {
+  if (
+    process.env.NODE_ENV !== 'production' &&
+    process.env.NODE_ENV !== 'test'
+  ) {
+    deprecate(
+      'NotificationList',
+      'The NotificationList component is deprecated.',
+      'Use the new Notification components instead',
```
Comment (connor-baer) [inline packages/circuit-ui/components/NotificationList/NotificationList.tsx:83]:
```suggestion
      'Use the new notification components instead.',
```

File: MIGRATION.md [hunk 5]
```diff
@@ -70,6 +71,38 @@ Tip: Provide the `--transform`/`-t` argument at the end of the command, so that
...
+### Notification components
+
+The old Notification components were general purpose without clear guidelines when to use a specific Notification component. This has led to inconsistent usage in our apps.
+The deprecated `Notification`, `NotificationList` and `NotificationCard` components were replaced with the new refreshed Notification components (`NotificationBanner`, `NotificationFullscreen`, `NotificationModal`, `NotificationToast` and `NotificationInline`)
+This introduces semantic, accessible components that make it clear when each should be used and are flexible enough to cover all use cases.
+
```
[the remark this answers is not in the document]
Comment (amelako) [reply MIGRATION.md]: Thanks a lot for the feedback. I shortened it a lot, let me know if it makes more sense, and should I add more info.

File: MIGRATION.md [hunk 6]
```diff
@@ -70,6 +72,22 @@ Tip: Provide the `--transform`/`-t` argument at the end of the command, so that
...
+The old Notification components were general purpose without clear guidelines when to use a specific Notification component. This has led to inconsistent usage in our apps.
+
+The deprecated `Notification`, `NotificationList` and `NotificationCard` components were replaced with the new refreshed Notification components (`NotificationBanner`, `NotificationFullscreen`, `NotificationModal`, `NotificationToast` and `NotificationInline`).
+This introduces semantic, accessible components that make it clear when each should be used and are flexible enough to cover all use cases.
+
+- The deprecated `InlineMessage` was replaced with the new `NotificationInline` component.
+- The deprecated `NotificationCard` component can be replaced with either `NotificationInline`, `NotificationBanner` or `NotificationToast` depending on use cases.
+
+Refer to [the Notification section in Storybook](https://circuit.sumup.com/?path=/docs/notification) for a usage examples of the new Notification components.
```
Comment (robinmetral) [inline MIGRATION.md:89]:
```suggestion
Refer to [the Notification section in Storybook](https://circuit.sumup.com/?path=/docs/notification) for usage examples of the new Notification components.
```

**Review by robinmetral — APPROVED** [review]
Looks good to me, thanks for handling the deprecations here!

**Pull request merged**